plugins/inputs/file/file.go

```diff
@@ -63,7 +63,10 @@ func (f *File) Gather(acc telegraf.Accumulator) error {
 			if f.FileTag != "" {
 				m.AddTag(f.FileTag, filepath.Base(k))
 			}
-			acc.AddFields(m.Name(), m.Fields(), m.Tags(), m.Time())
+			fmt.Printf("metricTime (unix): [%v] (%d)\n", m.Time(), m.Time().UnixNano())
+
+			acc.AddFields(m.Name(), m.Fields(), m.Tags())
```

Comment:
The problem is this line. AddFields takes an optional time argument, and if you don't supply it, it uses time.Now()

	// AddFields adds a metric to the accumulator with the given measurement
	// name, fields, and tags (and timestamp). If a timestamp is not provided,
	// then the accumulator sets it to "now".